BondageClub/Scripts/Dialog.js

```diff
@@ -1002,28 +1011,66 @@ function DialogFacialExpressionsLoad(Slot) {
 		DialogFacialExpressionsBuild();
 	}
 }
-/**draws the savedexpressions menu */
+
+/**
+ * Builds the savedexpressions menu previews.
+ * @returns {void} - Nothing
+ */
+function DialogBuildSavedExpressionsMenu() {
+	const ExcludedGroups = ["Hat", "Mask", "ClothAccessory", "Glasses"];
+	const BaseAppearance = Player.Appearance.filter(A => A.Asset.Group.Category === "Appearance" && !ExcludedGroups.includes(A.Asset.Group.Name));
+	Player.SavedExpressions.forEach((expression, i) => {
+		if (expression) {
+			const PreviewCharacter = CharacterLoadSimple("SavedExpressionPreview-" + i);
+			PreviewCharacter.Appearance = BaseAppearance.slice();
+			BaseAppearance.forEach(A => InventoryWear(PreviewCharacter, A.Asset.Name, A.Asset.Group.Name, A.Color));
+
+			for (let x = 0; x < expression.length; x++) {
+				CharacterSetFacialExpression(PreviewCharacter, expression[x].Group, expression[x].CurrentExpression);
+			}
+
+			DialogSavedExpressionPreviews[i] = PreviewCharacter;
+		}
+	});
+}
+
+/**
+ * Draws the savedexpressions menu
+ * @returns {void} - Nothing
+ */
 function DialogDrawSavedExpressionsMenu() {
-	DrawText(DialogFindPlayer("SavedExpressions"), 195, 25, "White", "Black");
-	DrawText(DialogFindPlayer("SavedExpressionsSave"), 140, 180, "White", "Black");
-	DrawText(DialogFindPlayer("SavedExpressionsLoad"), 260, 180, "White", "Black");
+	DrawText(DialogFindPlayer("SavedExpressions"), 210, 90, "White", "Black");
+
+	if (!DialogSavedExpressionPreviews || !DialogSavedExpressionPreviews.length)
+		DialogBuildSavedExpressionsMenu();
+
 	for (let x = 0; x < 5; x++) {
-		DrawButton(100, 200 + (x * 100), 80, 80, x + 1, "White");
-		DrawButton(220, 200 + (x * 100), 80, 80, x + 1, "White");
+		if (Player.SavedExpressions[x] == null) {
+			DrawText(DialogFindPlayer("SavedExpressionsEmpty"), 160, 216 + (x * 170), "White", "Black");
+		} else {
+			const PreviewCanvas = DrawCharacterSegment(DialogSavedExpressionPreviews[x], 100, 0, 300, 220);
+			MainCanvas.drawImage(PreviewCanvas, 0, 92 + (x * 170), 300, 190);
```

Comment:
```suggestion
			MainCanvas.drawImage(PreviewCanvas, 20, 92 + (x * 170), 260, 190);
```
The expression previews are noticeably stretched horizontally. I think they look much better in the original ratio.